wit-0.3.0-draft/terminal.wit

```diff
@@ -1,25 +1,358 @@
+    /// | `␛[2~`       | Insert                                               | key_ic    | kich1 | kI |
+    /// | `␛[3~`       | Delete                                               | key_dc    | kdch1 | kD |
+    /// | `␛[5~`       | Page Up                                              | key_ppage | kpp   | kP |
+    /// | `␛[6~`       | Page Down                                            | key_npage | knp   | kN |
+    /// | `␛OP`        | F1                                                   | key_f1    | kf1   | k1 |
+    /// | `␛OQ`        | F2                                                   | key_f2    | kf2   | k2 |
+    /// | `␛OR`        | F3                                                   | key_f3    | kf3   | k3 |
+    /// | `␛OS`        | F4                                                   | key_f4    | kf4   | k4 |
+    /// | `␛[15~`      | F5                                                   | key_f5    | kf5   | k5 |
+    /// | `␛[17~`      | F6                                                   | key_f6    | kf6   | k6 |
+    /// | `␛[18~`      | F7                                                   | key_f7    | kf7   | k7 |
+    /// | `␛[19~`      | F8                                                   | key_f8    | kf8   | k8 |
+    /// | `␛[20~`      | F9                                                   | key_f9    | kf9   | k9 |
+    /// | `␛[21~`      | F10                                                  | key_f10   | kf10  | k; |
+    /// | `␛[23~`      | F11                                                  | key_f11   | kf11  | F1 |
+    /// | `␛[24~`      | F12                                                  | key_f12   | kf12  | F2 |
+    /// | `␛[200~`     | Begin Paste; only emitted when bracketed paste mode is activated | PS | PS | \* |
+    /// | `␛[201~`     | End Paste; only emitted when bracketed paste mode is activated | PE | PS | \* |
+    ///
+    /// In immediate mode, the following additional control codes are defined
+    /// for input:
+    ///
+    /// Control codes:
+    ///
+    /// | Code | Meaning                                                      | terminfo  | ti    | tc |
+    /// | ---- | ------------------------------------------------------------ | --------- | ----- | -- |
+    /// | U+C  | Request to refresh the screen                                | key_refresh krfr  | &2 |
+    /// | U+1B | Escape                                                       | \*        | \*    | \* |
+    ///
+    /// In non-immediate mode, the following additional control codes are
+    /// defined for input:
+    ///
+    /// Control codes:
+    ///
+    /// | Code | Meaning                                                      | terminfo  | ti    | tc |
+    /// | ---- | ------------------------------------------------------------ | --------- | ----- | -- |
+    /// | U+D  | Enter                                                        | key_enter | kent  | @8 |
```

Comment:
shouldn't this be for immediate mode? in non-immediate mode generally unix terminals send `\n` for Enter.

Reply:
Oops, good catch. I was intending to describe exactly that rule, but got the sense of the table here backwards. I've now removed that line, and added a different laying saying that Ctrl-M is mapped to `\n` in non-immediate mode.